lib/Raven/Client.php

```diff
@@ -112,13 +113,26 @@ class Client
     protected $config;
 
     /**
+     * @var HttpAsyncClient The HTTP client
+     */
+    private $httpClient;
+
+    /**
+     * @var MessageFactory The message factory
+     */
+    private $messageFactory;
+
+    /**
      * Constructor.
      *
-     * @param Configuration $config The client configuration
+     * @param Configuration   $config     The client configuration
+     * @param HttpAsyncClient $httpClient The HTTP client
      */
-    public function __construct(Configuration $config)
+    public function __construct(Configuration $config, HttpAsyncClient $httpClient, MessageFactory $messageFactory)
```

Comment:
"Single parameter" is not a pattern. Also, it's not useful, and it will not have the benefits of typehints that we have here. If we will have too many parameters, it will mean that we are breaking the SRP principle and we would need to refactor a little.